### DISCUSSION TITLE
Why is int(int(...)) used in unpack_bitstring?

### DISCUSSION THREAD

**Discussion by MrWaloo** [pymodbus-dev/pymodbus#2314] [Q&A] [answered]
Hi,

I haven't made tons of tests, I just have watched the code and I wonder why `int(int( ... ))` is used here:
https://github.com/pymodbus-dev/pymodbus/blob/eb1d574841357dcc2784651ddff9e5be972902c5/pymodbus/utilities.py#L141

I doubt that this kind of workaround is needed in python3.9+. But I would like to have an expert input.

Thank you

**Comment by janiversen** [comment, marked as answer]
You might be right, but what is the problem? does the code not work ?

There are with security tons of lines that could be written more efficiently.

Please submit a pull request if you want it changed.
